apps/cms.c

```diff
@@ -1137,7 +1137,7 @@ int cms_main(int argc, char **argv)
             if (kparam != NULL) {
                 EVP_PKEY_CTX *pctx;
                 pctx = CMS_SignerInfo_get0_pkey_ctx(si);
-                if (!cms_set_pkey_param(pctx, kparam->param))
+                if (pctx != NULL && !cms_set_pkey_param(pctx, kparam->param))
```

Comment:
This is a wrong fix here. Correct condition would be `pctx == NULL || !cms_set_pkey_param(pctx, kparam->param)`